File: benchmark/c/cpw/check_number_is_perfect_square_using_additionsubtraction/check_number_is_perfect_square_using_additionsubtraction.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int check_number_is_perfect_square_using_additionsubtraction ( int n ) ;
int check_number_is_perfect_square_using_additionsubtraction ( int n ) {
  for ( int sum = 0, i = 1;
  sum < n;
  i += 2 ) {
    sum += i;
    if ( sum == n ) return 1;
  }
  return 0;
}

File: translation_result/else_bms/fail/c/cpw/check_number_is_perfect_square_using_additionsubtraction/gemini/BaseRepair/check_number_is_perfect_square_using_additionsubtraction.rs

fn check_number_is_perfect_square_using_additionsubtraction(n: i32) -> bool {
    let mut sum = 0;
    for i in (1..).step_by(2) {
        sum += i;
        if sum == n {
            return true;
        }
    }
    false
}
